pkg/adapters/goframe/middleware.go

```diff
@@ -0,0 +1,47 @@
+package goframe
+
+import (
+	"github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/gogf/gf/v2/net/ghttp"
+	"net/http"
+)
+
+// SentinelMiddleware returns new ghttp.HandlerFunc
+// Default resource name is {method}:{path}, such as "GET:/api/users/:id"
+// Default block fallback is returning 429 status code
+// Define your own behavior by setting options
+func SentinelMiddleware(opts ...Option) ghttp.HandlerFunc {
+	options := evaluateOptions(opts)
+	return func(r *ghttp.Request) {
+		resourceName := r.Method + ":" + r.URL.Path
+
+		if options.resourceExtract != nil {
+			extractedName := options.resourceExtract(r)
+			if extractedName == "" {
+				extractedName = resourceName
+			}
+			resourceName = extractedName
```

Comment:
if extractedName := options.resourceExtract(r); extractedName != "" {
  resourceName = extractedName
}
It looks the same as the code logic above? But readability will be improved